packages/cozy-realtime/README.md

```diff
@@ -38,86 +38,42 @@ A simple way to have realtime using Websocket with [cozy-stack](https://github.c
-    domain: 'cozy.tools:8080',
-    secure: true // to use wss (with SSL) or not
-  }
-const subscription = realtime.subscribe(config, 'io.mocks.mydocs')
-
-// your code when a new document is created
-subscription.onCreate(doc => doSomethingOnCreate(doc))
-// your code when a document is updated
-subscription.onUpdate(doc => doSomethingOnUpdate(doc))
-// your code when a document is deleted
-subscription.onDelete(doc => doSomethingOnDelete(doc))
-
-// Unsubscribe all events from realtime
-subscription.unsubscribe()
-
-// for a specific document
-const docSubscription = realtime.subscribe(config, 'io.mocks.mydocs')
-
-// There is no onCreate here since to have the id,
-// the document is already created
-
-// your code when your document is updated
-docSubscription.onUpdate(doc => doSomethingOnUpdate(doc))
-// your code when your document is deleted
-docSubscription.onDelete(doc => doSomethingOnDelete(doc))
-
-// Unsubscribe all events from realtime
-docSubscription.unsubscribe()
-```
-
-All `onCreate`, `onUpdate` and `onDelete` methods will return the subscription so you can chain the call like below:
-
-```javascript
-import realtime from 'cozy-realtime'
-
-const config = {
-    token: authToken, // app token provided by the stack or the client
-    domain: 'cozy.tools:8080',
-    secure: true // to use wss (with SSL) or not
-  }
-const subscription = realtime.subscribe(config, 'io.mocks.mydocs')
-  // your code when a new document is created
-  .onCreate(doc => doSomethingOnCreate(doc))
-  // your code when a document is updated
-  .onUpdate(doc => doSomethingOnUpdate(doc))
-  // your code when a document is deleted
-  .onDelete(doc => doSomethingOnDelete(doc))
-
-// Unsubscribe all events from realtime
-subscription.unsubscribe()
+## Example
+
+```js
+import CozyRealtime, { EVENT_CREATED, EVENT_UPDATED } from 'cozy-realtime'
+
+const realtime = new CozyRealtime({ cozyClient })
+const type = 'io.cozy.accounts'
+const id = 'document_id'
+const handlerCreate = accounts => {
+  console.log(`A new 'io.cozy.accounts' is created with id '${accounts._id}'.`)
+}
+const handlerUpdate = accounts => {
+  console.log(`An account is updated with id '${accounts._id}'.`)
+}
+
+// To subscribe
+await realtime.onCreate({ type }, handlerCreate)
+await realtime.onUpdate({ type, id }, handlerUpdate)
+
+// To unsubscribe event for a type, an event name and an handler
+await realtime.unsubscribe({ type, eventName: EVENT_CREATED }, handlerCreate)
+// To unsubscribe all events for a type, id and an event name
+await realtime.unsubscribe({ type, eventName: EVENT_UPDATED, id })
+// To unsubscribe all events for a type and an event name
+await realtime.unsubscribe({ type, eventName: EVENT_UPDATED })
+// To unsubscribe all event for a type and id
+await realtime.unsubscribe({ type, id })
+// To unsubscribe all event for a type
+await realtime.unsubscribe({ type })
+// To unsubscribe all
+await realtime.unsubscribeAll()
```

Comment:
Je suis pas fan de l'objet en premier argument. J'ai jamais vu ça et j'ai l'impression qu'il y aura des erreurs.

Je partirais plus sur une API plus classique sans objet, comme dans Backbone par exemple.

```
on('create', 'io.cozy.doctypes', handleCreate)
on('create', 'io.cozy.doctypes', myDocumentId, handleCreateMyDoc)
off('create', 'io.cozy.doctypes', handleCreate)

on('all', 'io.cozy.doctypes', handleAll)
off('all', 'io.cozy.doctypes', handleAll)
```

Reply:
I understand and I completely agree with this way to work if we have more event names. But that's not the case here and I am not comfortable about using a string as API here, the string can easily be subject to a typo somewhere we don't see it and we can not log the API to get all available methods on it.
Since those string won't change I would prefer some method like `onCreate/offCreate` instead IMHO

Reply:
Je préfère une signature de méthode constante, mais je comprends aussi les intérêts de la méthode Backbone ici. Donc, sans opinion.

Reply:
> the string can easily be subject to a typo somewhere we don't see it a

This is a good point, the `subscribe` method can easily do

```jsx 
const subscribe = (eventName, ...) => {
  if (!validEvents.include(eventName)) {
    throw new Error(`${eventName} is not a valid event, valid events are ${validEvents}`)
  }
}